modules/gradio_ui.py

```diff
@@ -2,61 +2,73 @@
 import requests
 import json
 
-conversation_id=None
+class gradioUI:
+    def __init__(self,
+                 ols_url="http://127.0.0.1:8080/ols",
+                 mount_path="/ui",
+                 conversation_id=None) -> None:
+        # class variable
+        self.ols_url=ols_url
+        self.mount_path=mount_path
+        self.conversation_id = conversation_id
```

Comment:
orthogonal to your PR but something I was wondering about....  there are multiple users hitting the gradio UI, do they each get their own instance of this class?  If not, the "conversation_id" would be shared between the two users which would be bad.

Not super critical for a debug ui, but something i'd like to understand about gradio in general if you happen to know. (And if nothing else, if it is indeed going to be shared then we should at least understand that constraint).

Same thing for the "use_history" config value/checkbox.

Reply:
They should, but it is actually a little bit more complicated than that. The uvicorn execute a thread per session per endpoint. (each request is handle on a separate thread). In that case each thread accessing the UI expoing will have its own settings. Each one will get its own copy of everything in main app (ols.py) that it is needed for that "def" function on FastAPI and these are processed async. When aggregating endpoints in other ways (e.g. like route aggregation), the behavior can be different.

Reply:
uvicorn having a thread per session doesn't tell me whether the gradio instance is shared between those threads, though...

i haven't tested w/ your refactor, but i just tested the code from main and indeed the conversation id is shared between two browser sessions (I started one in chrome and one in FF).  So at least in the implementation in main, there is only a single "Gradio" object instance, with its associated "conversation id" field, and once it's set by one interaction in one browser, it gets reused by interactions in the other browser.

again, not critical for a debugging tool right now, but important to understand (and probably has implications for other spots where we might be using class instance variables and the class is shared between requests/sessions)

Reply:
A quick test using two browser sessions seems they are independent but this needs better testing